src/os/bluestore/FreelistManager.cc

```diff
@@ -199,7 +203,20 @@ int FreelistManager::release(
 	       << " (merge with previous)" << dendl;
       length += p->second;
       offset = p->first;
-      kv_free.erase(p++);
+      if (map_t_has_stable_iterators) {
+	kv_free.erase(p++);
+      } else {
+	uint64_t next = 0;
+	++p;
+	if (p != kv_free.end())
+	  next = p->first;
+	--p;
+	kv_free.erase(p);
```

Comment:
It looks like btree_map::erase returns an iterator for the next item:
See btree_map_container declaration:
...
// Erase the specified iterator from the btree. The iterator must be valid
// (i.e. not equal to end()).  Return an iterator pointing to the node after
// the one that was erased (or end() if none exists).
iterator erase(const iterator &iter) {
   return this->tree_.erase(iter);
}
...

Thus you probably can have something like 
p = kv.free.erase(p);
and remove other stuff to locate the next entry